vowpalwabbit/cb_explore_adf.cc

```diff
@@ -146,11 +150,13 @@ void predict_or_learn_greedy(cb_explore_adf& data, base_learner& base, v_array<e
     multiline_learn_or_predict<false>(base, examples, data.offset);
 
   v_array<action_score>& preds = examples[0]->pred.a_s;
-  uint32_t num_actions = (uint32_t)preds.size();
-  float prob = data.epsilon/(float)num_actions;
-  for (size_t i = 0; i < num_actions; i++)
-    preds[i].score = prob;
-  preds[0].score += 1.f - data.epsilon;
+
+  // generate distribution over actions
+  vector<float> pdf(preds.size());
```

Comment:
Move this variable into cb_explore_adf?  This avoids both redeclaration across multiple reductions and stack allocation in the reduction.

Reply:
vector doesn't support the lazy shrink like v_array does. You want me to switch to v_array or just move the vector and reallocate every time (the later is slightly more annoying as I need to deal with explicit ctor/dtor calls)

Reply:
I don't believe we need a lazy shrink here, so you can move the vector to the reduction and just reuse it without reallocation.